### DISCUSSION TITLE
How do I setup?

### DISCUSSION THREAD

**Discussion by baum1234562** [cantalo/cantalo#17] [Q&A] [answered]
Hey,
So I am quite new to svelteKit and in general web programming. I want to do something similar to a karaoke website so I wanted to understand your code in order to write my own. But I dont know how I setup your code in order that it works as a website I always get only a blackscreen.

**Comment by bbuhler** [comment, marked as answer]
Hi @baum1234562,
to run the application you need to define a data source. Lately I was experimenting with different data sources. Depends when you checked out the project, you need either a LowDB JSON file with a songs collection (defined in environment variables as `DB_PATH`) or with recent changes a web service (defined in environment varibales as `SONG_SERVICE`) responding with a song index JSON and song data JSON (see `src/lib/server/db.js`).

Here is an example for a song data JSON:
[GUl9_5kK9ts.json](https://github.com/cantalo/cantalo/files/14974485/GUl9_5kK9ts.json)

The song index JSON consist of a collection of properties from `meta`.
